File: LifeSamadhan/Backend/Models/Service.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeSamadhan.API.Models
{
    [Table("services")]
    public class Service
    {
        [Key]
        [Column("Id")]
        public long Id { get; set; }

        [Required(ErrorMessage = "CategoryId is required")]
        [Column("CategoryId")]
        public long CategoryId { get; set; }

        [ForeignKey(nameof(CategoryId))]
        public ServiceCategory? Category { get; set; }

        [Required(ErrorMessage = "Service name is required")]
        [StringLength(150, MinimumLength = 3,
            ErrorMessage = "Service name must be between 3 and 150 characters")]
        [Column("Name")]
        public string Name { get; set; } = string.Empty;

        [RegularExpression("^(ACTIVE|INACTIVE)$",
            ErrorMessage = "Status must be ACTIVE or INACTIVE")]
        [Column("Status")]
        public string Status { get; set; } = "ACTIVE";
    }
}

File: lifeFull/SpringBootLifeSamadhan/src/main/java/com/lifesamadhan/api/model/Service.java
package com.lifesamadhan.api.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Entity
@Table(name = "services", uniqueConstraints = @UniqueConstraint(columnNames = { "category_id", "name" }))
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Service {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "CategoryId is required")
    @Min(value = 1, message = "CategoryId must be a valid value")
    @Column(name = "category_id", nullable = false)
    private Long categoryId;

    @NotBlank(message = "Service name is required")
    @Size(min = 3, max = 150, message = "Service name must be between 3 and 150 characters")
    @Column(nullable = false, length = 150)
    private String name;

    @Pattern(regexp = "^(ACTIVE|INACTIVE)$", message = "Status must be ACTIVE or INACTIVE")
    @Builder.Default
    @Column(nullable = false)
    private String status = "ACTIVE";

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", insertable = false, updatable = false)
    @JsonIgnore
    private ServiceCategory category;

    @OneToMany(mappedBy = "service", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    @JsonIgnore
    private java.util.List<ProviderSkill> providerSkills;

    @OneToMany(mappedBy = "service", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    @JsonIgnore
    private java.util.List<ServiceRequest> serviceRequests;
}